### DISCUSSION TITLE
Continuous operation bucket / Incident events

### DISCUSSION THREAD

**Discussion by m-linner-ericsson** [cdfoundation/sig-events#49] [General]
In our current suggestion for the protocol, we have not defined any events or buckets with regards to operations and issue handling. 

This limits the use of the protocol and hinders players such as [4Keys](https://github.com/GoogleCloudPlatform/fourkeys) to use it. 4Keys uses these issue events to track how long time it takes to fix an issue and how many issues you get per release/deployment.

As an example of this type of event: In Eiffel, we can link/reference our Source Code Version Control Events to an  [Issue Defined Event](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelIssueDefinedEvent.md). This enables us to track what bugs and where we fixed them.

I propose adding such types of events to our protocol for tracking issues and connecting them to the rest of the process.

What is your opinion?

**Comment by erkist** [comment]
I am absolutely in favor of adding incident events. In a continuous delivery process we are likely to discover errors not only during regular test automation, so it is not enough to just be reporting failed tests (and, even if we report failed tests, reporting a new issue, or connecting to an existing issue, is a typically a separate step).

When it comes to operation events, I would like be interested to hear more about it. For instance, how do they differ from the service-related events in continuous deployment? And are there any Eiffel events you would consider to be connected to operation?

**Reply by m-linner-ericsson** [reply]
My idea with introducing an operations bucket was to capture events like Issue Defined but the naming might not be the best. At one point I heard about CI/CD/CO whereas CO is continuous operations where the issue event could fit. When looking at a DevOps picture e.g. https://devopedia.org/images/article/54/7602.1513404277.png the "operate, monitor and plan" are not captured by our current buckets as I see. To capture those phases an operations bucket might help. However, naming is hard so I am open to better suggestions. Continuous Feedback maybe?
